newIDE/app/src/Profile/Subscription/CancelReasonDialog.js

```diff
@@ -0,0 +1,270 @@
+  const [isCancelingSubscription, setIsCancelingSubscription] = React.useState(
+    false
+  );
+  const [hasCanceledSubscription, setHasCanceledSubscription] = React.useState(
+    false
+  );
+  const [
+    stoppedMakingGamesChecked,
+    setStoppedMakingGamesChecked,
+  ] = React.useState(false);
+  const [strugglingChecked, setStrugglingChecked] = React.useState(false);
+  const [
+    preferFreeVersionChecked,
+    setPreferFreeVersionChecked,
+  ] = React.useState(false);
+  const [missingFeatureChecked, setMissingFeatureChecked] = React.useState(
+    false
+  );
+  const [qualityIssuesChecked, setQualityIssuesChecked] = React.useState(false);
+  const [otherChecked, setOtherChecked] = React.useState(false);
+  const [freeText, setFreeText] = React.useState('');
+  const authenticatedUser = React.useContext(AuthenticatedUserContext);
+  const { showAlert } = useAlertDialog();
+
+  const canSubmit = freeText.trim().length > 0;
+
+  const cancelPlan = React.useCallback(
+    async () => {
+      if (isCancelingSubscription || !canSubmit) return;
+      const {
+        getAuthorizationHeader,
+        subscription,
+        profile,
+      } = authenticatedUser;
+      if (!profile || !subscription) return;
+      setIsCancelingSubscription(true);
+      try {
+        await changeUserSubscription(
+          getAuthorizationHeader,
+          profile.id,
+          {
+            planId: null,
+          },
+          {
+            cancelImmediately: false,
+            cancelReasons: {
+              'stopped-making-games': stoppedMakingGamesChecked,
+              'struggling-with-gdevelop': strugglingChecked,
+              'prefer-free-version': preferFreeVersionChecked,
+              'missing-feature': missingFeatureChecked,
+              'quality-issues': qualityIssuesChecked,
+              other: otherChecked,
+              freeText: freeText,
+            },
+          }
+        );
+        await authenticatedUser.onRefreshSubscription();
+        setHasCanceledSubscription(true);
+      } catch (rawError) {
+        await authenticatedUser.onRefreshSubscription();
+        console.error(rawError);
+        showAlert({
+          title: t`Could not cancel your subscription`,
+          message: t`There was an error while canceling your subscription. Verify your internet connection or try again later.`,
+        });
+      } finally {
+        setIsCancelingSubscription(false);
+      }
+    },
+    [
+      authenticatedUser,
+      showAlert,
+      isCancelingSubscription,
+      canSubmit,
+      freeText,
+      stoppedMakingGamesChecked,
+      strugglingChecked,
+      preferFreeVersionChecked,
+      qualityIssuesChecked,
+      missingFeatureChecked,
+      otherChecked,
+    ]
+  );
+
+  const isLoading =
+    !authenticatedUser.subscription ||
+    !authenticatedUser.profile ||
+    isCancelingSubscription;
+
+  const actions = hasCanceledSubscription
```

Comment:
Unsure if better, but I see a lot of if `hasCanceledSubscription` - could almost have been a single `if (hasCanceledSubscription) return <Dialog`.

Reply:
Yeah I think it will recreate a dialog which may look weird, like blinking, so unsure

Reply:
It should not, it's the same JSX/React elements from the point of view of React :) 
Anyway, not a big deal.